src/components/RoleCardsContainer/RoleCard/index.jsx

```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import {
+  Card,
+  CheckIcon,
+  IconContainer,
+  Overview,
+  Rectangle,
+  RoleText,
+} from './role-card.styles';
+
+/**
+ * Role Card.
+ *
+ * @param {Object} props                 Props passed to RoleCard.
+ * @param {Object} props.Icon            Role's icon.
+ * @param {string} props.overview        Role's overview.
+ * @param {string} props.role            Role's title.
+ * @param {func}   props.setSelectedRole Sets selected role.
+ *
+ * @returns {JSX.Element} Role Card holding role details.
+ */
+function RoleCard({ Icon, overview, role, setSelectedRole }) {
+  const [isClicked, setIsClicked] = useState(false);
+
+  const handleCardSelect = () => {
+    setSelectedRole((prev) => new Set(prev).add(role));
+    if (isClicked)
+      setSelectedRole((prev) => {
+        const next = new Set(prev);
+        next.delete(role);
+        return next;
+      });
+    setIsClicked(!isClicked);
+  };
+
+  return (
+    <Card onClick={handleCardSelect} clicked={isClicked}>
+      <Rectangle clicked={isClicked}>{isClicked && <CheckIcon />}</Rectangle>
+      <IconContainer>
+        <Icon clicked={isClicked ? 'true' : undefined} />
+      </IconContainer>
+      <RoleText>{role}</RoleText>
+      <Overview>{overview}</Overview>
+    </Card>
+  );
+}
+
+RoleCard.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```

Comment:
can we git rid of this eslint disabling line here, 
```suggestion
 Icon: PropTypes.oneOf(['img', 'svg']).isRequired,
```